gengokit/template/NAME-service/cmd/NAME/main.gotemplate

```diff
@@ -101,6 +119,15 @@ func submain() int {
 		return 1
 	}
 
+	if len(flag.Args()) < 1 {
+		flag.Usage()
+		return 1
+	}
+
+	ctx := context.Background()
+	for i := 0; i < len(headers); i += 2 {
+		ctx = context.WithValue(ctx, headers[i], headers[i+1])
+	}
 
 	switch flag.Args()[0] {
```

Comment:
It would be great to switch this out with
```
args := flag.Args()
if len(args) < 1 {
    // print message about needing method
    os.Exit(1)
}

So it does not panic if flags are provided but not arguments.